kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/DataClassEq.kt

```diff
@@ -64,9 +65,12 @@ internal object DataClassEq : Eq<Any> {
             dataClassDiff(actualPropertyValue, expectedPropertyValue, depth + 1)?.let { diff ->
                Pair(prop, diff)
             }
-         else
-            runCatching { actualPropertyValue shouldBe expectedPropertyValue }
-               .fold(onSuccess = { null }, onFailure = { t -> Pair(prop, StandardDifference(t)) })
+         else {
+            eq(actualPropertyValue, expectedPropertyValue)
+               .toOption()
+               .map { Pair(prop, StandardDifference(it)) }
+               .orNull()
+         }
```

Comment:
Can you explain what the change is doing ?

Reply:
Earlier we are using `actualPropertyValue shouldBe expectedPropertyValue`

shouldBe internally add the error to errorCollector due to which we are getting multiple error message when data class contains enum, now i replaced that with eq function call which check equality and return error using which we are creating the difference message for data class without adding those extra errors in errorCollector.

Reply:
Is it good to merge?

Reply:
Ok I see.